### DISCUSSION TITLE
Detecting clients connected to AP, should isconnected() from WLAN tell me this?

### DISCUSSION THREAD

**Discussion by willlahr** [micropython/micropython#10688] [RP2040 / Pico]
I've got a Pi Pico W set up as an access point.  

It says here https://docs.micropython.org/en/latest/library/network.WLAN.html "In AP mode returns True when a station is connected." 

I'm reading this to say that when the AP is advertising but no devices have joined the wireless network it created the function should return False, and when (e.g.) I connect to it with my laptop it should return True

What I'm actually seeing is it return True in all cases.   Has anyone else seen this behaviour?  Could it be a bug - I couldn't find it mentioned in issues

**Comment by peterhinch** [comment]
Even in station mode `isconnected()` only reports that the device has connected to a LAN. It conveys no information about servers or other connected devices, merely that a network exists. I think the same applies in AP mode. If you are running a server on your AP, keeping a count of connected clients should be easily done.

**Comment by Tangerino** [comment]
A code snippet that can give you a hint
```python
async def handle_wifi_clientes(nic, current_clients: list) -> list:
    clients = nic.status('stations')
    if clients != current_clients:
        util.nic_wifi.clients = []
        if len(clients) == 0:
            log("[WIFI] No stations connected")
        else:
            log(f"[WIFI] {len(clients)} stations connected")
            for idx, mac in enumerate(clients, start=1):
                for m in mac:
                    smac = f"{m[0]:02x}:{m[1]:02x}:{m[2]:02x}:{m[3]:02x}:{m[4]:02x}:{m[5]:02x}"
                    log(f"[WIFI] Station `{idx}` connected: {smac}")
                    util.nic_wifi.clients.append(smac)
    return clients
```
